standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -708,6 +759,13 @@ private void removeTxnComponents(Connection dbConn, CompactionInfo info) throws
     PreparedStatement pStmt = null;
     ResultSet rs = null;
     try {
+      /*
+       * Remove all abort retry associated metadata of table/partition in the COMPACTION_QUEUE both when compaction
+       * or abort cleanup is successful. We don't want a situation wherein we have a abort retry entry for a table
+       * but no corresponding entry in TXN_COMPONENTS table.
+       */
+      removeAbortRetryEntries(dbConn, info);
```

Comment:
not required, we should do the cleanup per cd_id

Reply:
As discussed, this should be done according to db, table and partition names only due to compaction. Done.